trusted/crypto/src/lib.rs

```diff
@@ -65,4 +156,34 @@ mod tests {
         };
         assert_eq!(hash.as_slice(), &expected[..]);
     }
+
+    #[test]
+    fn build_sha256_1984() {
+        let bytes: [u8; 4] = [1, 9, 8, 4];
+        let mut builder = Sha256Builder::new().unwrap();
+        for i in bytes {
+            builder = builder.update(&[i]).unwrap();
+        }
+        let hash = builder.build().unwrap();
+        let expected = {
+            let mut hasher = sha2::Sha256::new();
+            hasher.update(&bytes);
+            hasher.finalize()
+        };
```

Comment:
Consider adding a helper (in this mod) for this pattern

Reply:
done